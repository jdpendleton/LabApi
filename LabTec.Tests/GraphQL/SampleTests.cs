using HotChocolate.Execution;
using LabTec.GraphQL.Services;
using LabTec.Infrastructure.Extensions;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;

namespace LabTec.Tests.GraphQL
{
    public class SampleTests
    {
        [Fact]
        public async Task SchemaChangeTest()
        {
            #region Arrange



            #endregion

            #region Act

            var schema = await new ServiceCollection()
                .AddGraphQLServer()
                .AddQueryType(d => d.Name("Query"))
                    .AddType<SampleQueries>()
                .AddMutationType(d => d.Name("Mutation"))
                    .AddType<SampleMutations>()
                .BuildSchemaAsync();

            #endregion

            #region Assert

            schema.ToString().MatchSnapshot();

            #endregion
        }

        [Fact]
        public async Task CreateSample_Snapshot()
        {
            #region Arrange

            var query = $@"
                mutation {{
                    createSample (
                        request: {{
                            name: ""a4dac2d0-eda7-4a47-aabc-809b201364aa""
                            owner: ""a4dac2d0-eda7-4a47-aabc-809b201364aa""
                            position: 1 }}
                        ) {{
                        sample {{
                            name
                        }}
                    }}
                }}";

            #endregion

            #region Act

            var result = await new ServiceCollection()
                .AddInfrastructureServices()
                .AddSingleton<SampleService>()
                .AddGraphQLServer()
                .AddQueryType(d => d.Name("Query"))
                    .AddType<SampleQueries>()
                .AddMutationType(d => d.Name("Mutation"))
                    .AddType<SampleMutations>()
                .ExecuteRequestAsync(query);

            #endregion

            #region Assert

            result.ToJson().MatchSnapshot();

            #endregion
        }
    }
}