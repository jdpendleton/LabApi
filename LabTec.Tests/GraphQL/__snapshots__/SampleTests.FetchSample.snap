{
  "data": {
    "createSample": {
      "sample": {
        "name": "a4dac2d0-eda7-4a47-aabc-809b201364aa"
      }
    }
  }
}
